bundles/io/org.eclipse.smarthome.io.rest/src/main/java/org/eclipse/smarthome/io/rest/LocaleServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.rest;
+
+import java.util.Locale;
+
+import org.eclipse.smarthome.core.i18n.LocaleProvider;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link LocaleServiceImpl} provides helper method for working with locales in REST
+ * resources.
+ *
+ * @author Dennis Nobel - Initial contribution
+ * @author Markus Rathgeb - Use locale provider
+ * @author Martin Herbst - Support of different language definition variants
+ * @authro Lyubomir Papazov - Add component annotation, rename the class to LocaleService and add method tryGetLocale
+ */
+@Component
+public class LocaleServiceImpl implements LocaleService {
+
+    private static final Logger logger = LoggerFactory.getLogger(LocaleServiceImpl.class);
+
+    private static LocaleProvider localeProvider;
```

Comment:
* remove `static`
* add `volatile`

Reply:
Do you claim `volatile` here because `localeProvider` is a dynamic reference and can be changed any time? Other threads holding a LocaleService need to check with the main copy then, correct?

Reply:
Correct.
For dynamic references volatile or atomic references should be used.
If you want to spend your time in reading some discussion about: https://github.com/eclipse/smarthome/pull/3617
:wink: